public/apps/configuration/panels/role-edit.tsx

```diff
@@ -47,24 +58,191 @@ const TITLE_TEXT_DICT = {
   create: 'Create Role',
   edit: 'Edit Role',
   duplicate: 'Duplicate Role',
-}
+};
 
 type OptionSeletion = EuiComboBoxOptionOption[];
 
-function buildPermissionOptions(optionsList: string[]) {
-  return optionsList.map(e => ({ label: e }));
+interface RoleIndexPermissionStateClass {
+  indexPatterns: OptionSeletion;
+  dls: string;
+  flsMethod: 'exclude' | 'include';
+  fls: OptionSeletion;
+  maskedFields: OptionSeletion;
+  allowedActions: OptionSeletion;
+}
+
+function getEmptyIndexPermission(): RoleIndexPermissionStateClass {
+  return {
+    indexPatterns: [],
+    dls: '',
+    flsMethod: 'exclude',
+    fls: [],
```

Comment:
could we use fullname for readability?

Reply:
This is the name same as in backend storage. But good point, I'll update the front end side to increase readability

Reply:
Updated